qutip/tests/test_mkl.py

```diff
@@ -31,205 +31,120 @@
-    out = mkl_spsolve(A,b)
-    assert_(b.shape==out.shape)
-
-
-@unittest.skipIf(qset.has_mkl == False, 'MKL extensions not found.')
-def test_mkl_spsolve8():
-    """
-    MKL spsolve : Sparse RHS matrix
-    """
-    A = sp.csr_matrix([
-                [1, 2, 0],
-                [0, 3, 0],
-                [0, 0, 5]])
-    b = sp.csr_matrix([
-        [0, 1],
-        [1, 0],
-        [0, 0]])
-
-    x = mkl_spsolve(A, b)
-    ans = np.array([[-0.66666667, 1],
-                    [0.33333333, 0],
-                    [0, 0]])
-    assert_array_almost_equal(x.toarray(), ans)
-
-
-@unittest.skipIf(qset.has_mkl == False, 'MKL extensions not found.')
-def test_mkl_spsolve9():
-    """
-    MKL spsolve : Hermitian (complex) solver
-    """
-    A = rand_herm(np.arange(1,11)).data
-    x = np.ones(10,dtype=complex)
-    b = A.dot(x)
-    y = mkl_spsolve(A, b, hermitian=1)
-    assert_array_almost_equal(x, y)
-    
-
-@unittest.skipIf(qset.has_mkl == False, 'MKL extensions not found.')
-def test_mkl_spsolve10():
-    """
-    MKL spsolve : Hermitian (real) solver
-    """
-    A = rand_herm(np.arange(1,11)).data
-    A = sp.csr_matrix((np.real(A.data), A.indices, A.indptr), dtype=float)
-    x = np.ones(10, dtype=float)
-    b = A.dot(x)
-    y = mkl_spsolve(A, b, hermitian=1)
-    assert_array_almost_equal(x, y)
-
-
-if __name__ == "__main__":
-    run_module_suite()
+import scipy.linalg
+import scipy.sparse
+
+import qutip
+if qutip.settings.has_mkl:
+    from qutip._mkl.spsolve import mkl_splu, mkl_spsolve
+
+pytestmark = [
+    pytest.mark.skipif(not qutip.settings.has_mkl,
+                       reason='MKL extensions not found.'),
+]
+
+
+class Test_spsolve:
+    def test_single_rhs_vector_real(self):
+        Adense = np.array([[0, 1, 1],
+                           [1, 0, 1],
+                           [0, 0, 1]])
+        As = scipy.sparse.csr_matrix(Adense)
+        np.random.seed(1234)
+        x = np.random.randn(3)
+        b = As * x
+        x2 = mkl_spsolve(As, b, verbose=True)
+        np.testing.assert_allclose(x, x2)
+
+    def test_single_rhs_vector_complex(self):
+        A = qutip.rand_herm(10)
+        x = qutip.rand_ket(10).full()
+        b = A.full() @ x
+        y = mkl_spsolve(A.data, b, verbose=True)
+        np.testing.assert_allclose(x, y)
+
+    @pytest.mark.parametrize('dtype', [np.float64, np.complex128])
+    def test_multi_rhs_vector(self, dtype):
+        M = np.array([
+            [1, 0, 2],
+            [0, 0, 3],
+            [-4, 5, 6],
+        ], dtype=dtype)
+        sM = scipy.sparse.csr_matrix(M)
+        N = np.array([
+            [3, 0, 1],
+            [0, 2, 0],
+            [0, 0, 0],
+        ], dtype=dtype)
+        sX = mkl_spsolve(sM, N, verbose=True)
+        X = scipy.linalg.solve(M, N)
+        np.testing.assert_allclose(X, sX)
+
+    def test_rhs_shape_is_maintained(self):
+        """
+        MKL spsolve : Solution shape same as input RHS vec
+        """
```

Comment:
Why keep the doc-string only here?

Reply:
A mistake - in my editor I have docstrings folded unless I open them, and I just overlooked it.  The docstrings don't add any information that we don't get from the names of the tests, which is why I removed them.